homeassistant/components/fritzbox/binary_sensor.py

```diff
@@ -1,27 +1,26 @@
 """Support for Fritzbox binary sensors."""
-import logging
-
 import requests
 
-from homeassistant.components.binary_sensor import BinarySensorDevice
-
-from . import DOMAIN as FRITZBOX_DOMAIN
+from homeassistant.components.binary_sensor import DOMAIN, BinarySensorDevice
+from homeassistant.const import CONF_DEVICES, CONF_ENTITIES
 
-_LOGGER = logging.getLogger(__name__)
+from .const import DOMAIN as FRITZBOX_DOMAIN, LOGGER
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Fritzbox binary sensor platform."""
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Fritzbox binary sensor from config_entry."""
     devices = []
-    fritz_list = hass.data[FRITZBOX_DOMAIN]
+    fritz_list = hass.data[FRITZBOX_DOMAIN][CONF_DEVICES]
+    entities = hass.data[FRITZBOX_DOMAIN][CONF_ENTITIES].setdefault(DOMAIN, set())
 
     for fritz in fritz_list:
         device_list = fritz.get_devices()
         for device in device_list:
-            if device.has_alarm:
+            if device.has_alarm and device.ain not in entities:
```

Comment:
If we only set up the devices for one fritz instance, may there still be a case of overlapping device ids (ain)?

Reply:
Yes, in my case I've a FRITZ!Box router and a FRITZ!PowerLine adapter. And both of them can turn the switch in der PowerLine adapter on/off.
Storing the fritz connection in the config_entry was too simple. Not even thought about it...

Reply:
We can't store the fritz instance directly in the config entry but we can use the config entry id as key in hass.data.